concent_api/concent_golem_messages/message.py

```diff
@@ -0,0 +1,133 @@
+import enum
+
+from golem_messages.message.base import Message
+from golem_messages.register import library
+from golem_messages.validators import validate_integer
+from golem_messages.validators import validate_varchar
+
+from concent_golem_messages.validators import validate_bytes
+
+
+CONCENT_CUSTOM_MSG_BASE = 9000
+
+
+class NonceAbstractMessage(Message):
+    """
+    Abstract message containing `nonce` field and its validation.
+    """
+
+    __slots__ = [
+        'nonce',
+    ] + Message.__slots__
+
+    def deserialize_slot(self, key, value):
+        value = super().deserialize_slot(key, value)
+        if key == 'nonce':
+            validate_integer(field_name=key, value=value)
+        return value
+
+
+class TransactionAbstractMessage(NonceAbstractMessage):
+    """
+    Abstract message containing transaction data and its validation.
+    """
+
+    __slots__ = [
+        'gasprice',
+        'startgas',
+        'to',
+        'value',
+        'data',
+    ] + NonceAbstractMessage.__slots__
+
+    def deserialize_slot(self, key, value):
+        value = super().deserialize_slot(key, value)
+        if key == 'to':
+            validate_varchar(
+                field_name=key,
+                value=value,
+                max_length=20,
+            )
+        if key == 'data':
+            validate_bytes(
+                field_name=key,
+                value=value,
+            )
+        if key in ('gasprice', 'startgas', 'value'):
+            validate_integer(
+                field_name=key,
+                value=value
```

Comment:
@kbeker https://github.com/golemfactory/concent/pull/687#discussion_r207895899
> The same situation as in NonceAbstractMessage we can add here validation for gasprice, startgas and value that it could have maximum 78 numbers

:+1: done